internal/hcs/process.go

```diff
@@ -290,6 +290,12 @@ func (process *Process) StdioLegacy() (_ io.WriteCloser, _ io.ReadCloser, _ io.R
 		return nil, nil, nil, makeProcessError(process, operation, ErrAlreadyClosed, nil)
 	}
 
+	if process.stdin != nil && process.stdout != nil && process.stderr != nil {
```

Comment:
Not positive, but I think there can be cases where only some of these are set (e.g. if Docker does not request stdin). So maybe these should be `||`.

Reply:
And I guess there's a problem where if Docker asks for no std handles, this will still fall through unnecessary, possibly leading to the same race condition we are trying to fix.

Maybe cleaner to add a bool and check that.

Reply:
I agree, I think just tracking via a bool is cleaner.

Reply:
Tracked via bool now. :)